kortex_driver/src/hardware_interface.cpp

```diff
@@ -758,8 +758,20 @@ return_type KortexMultiInterfaceHardware::read(
 {
   if (first_pass_)
   {
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
     first_pass_ = false;
-    feedback_ = base_cyclic_.RefreshFeedback();
+    try
+    {
+      feedback_ = base_cyclic_.RefreshFeedback();
+    }
+    catch (k_api::KDetailedException & ex)
+    {
+      RCLCPP_ERROR_STREAM(LOGGER, "Kortex exception during read first pass: " << ex.what());
+
+      RCLCPP_ERROR_STREAM(
+      LOGGER, "Error sub-code: " << k_api::SubErrorCodes_Name(
+       	         k_api::SubErrorCodes((ex.getErrorInfo().getError().error_sub_code()))));
```

Comment:
Will this help us know which type of fault was thrown to stop the arm motion? That would be exciting.

Reply:
not really this is mostly what kortex api exception was thrown, seems mostly related to communication issues.